### ISSUE TITLE
The logic of checking 'PUBLIC' and 'EXCEPT' is right?

### ISSUE THREAD

**Issue by aisuhua** [calinrada/PhalconUserPlugin#30]
I check the code of "Phalcon\UserPlugin\Plugin\Security.php". And I find the logic code of checking authority is not right. Below are code fragments:

	/**
     * Check for public resources.
     *
     * @param array  $resources
     * @param string $actionName
     * @param string $controllerName
     *
     * @return bool
     */
    private function checkPublicResources($resources, $actionName, $controllerName)
    {
        $resources = isset($resources['*']) ? $resources['*'] : $resources;
        foreach ($resources as $controller => $actions) {
            if ($controller == $controllerName) {
                if (isset($controller['*'])) {
                    return true;
                } else {
                    if (in_array($actionName, $actions) || $actions[0] == '*') {
                        return true;
                    }
                }
            }
        }
        return false;
    }

The doc says the  '*' meaning all except, but the code can not reach the goal. The problem the same as 'PRIVATE' AND 'EXCEPT'.

> In the exampe bellow, you will treat your website as public, EXCEPT the actions ACCOUNT and PROFILE from the USER controller

    'pup' => array(
        'redirect' => array(
            'success' => 'user/profile',
            'failure' => 'user/login'
        ),
        'resources' => array(
            'type' => 'public',
            'resources' => array(
                '*' => array( // All except
                    'user' => array('account', 'profile')
                ),
            )
         )
    )

Think ! So sorry my bad english.

**Comment by calinrada** [comment]
It is working because the  "$resources" param assign to checkPublicResources() method, it is $var['pup']['resources']['resources']

**Comment by aisuhua** [comment]
Yes, I know. But the `checkPublicResources()` return `true` is meaning "private resource". For example, if i visit the url `http://domain/user/login`, and  the config is:

	'pup' => array(
	    'redirect' => array(
	        'success' => 'user/profile',
	        'failure' => 'user/forbidden'
	    ),
	    'resources' => array(
	        'type' => 'public',
	        'resources' => array(
	            'user' => array('login', 'register')
	        )
	    )
	)

Now, as the doc says, the action `login` is public. But it redirect to `user/forbidden`. Is it right ?
